Performance/perf_runner.sh

```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+
+# SwiftProtobuf/Performance/perf_runner.swift - Performance test runner
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# -----------------------------------------------------------------------------
+#
+# This script creates a .proto file with the requested number of fields of a
+# particular type and generates a test harness that populates it, serializes
+# it, and then parses it back out. Then it generates the Swift source for the
+# .proto file, builds the harness, and runs it under Instruments to collect
+# timing samples.
+#
+# The results are stored in the Performance/Results directory. The instruments
+# traces are named based on the number and type of fields; multiple runs with
+# the same inputs will append to the existing trace file if it is present.
+#
+# In addition to the collected samples, this script also prints the time it
+# takes to compile/link the harness and the byte size of the harness before and
+# after stripping.
+#
+# -----------------------------------------------------------------------------
+
+set -eu
+
+function usage() {
+  cat >&2 <<EOF
+Usage: $0 [-p <true|false>] [-s2|-s3] <field count> <field type>
+
+Currently supported field types:
+    int32, sint32, uint32, fixed32, sfixed32,
+    int64, sint64, uint64, fixed64, sfixed64,
+    float, double, string,
+    ...and repeated variants of the above.
+
+Options:
+    -p <true|false>: Adds a packed option to each field.
+    -s[23]:          Generate proto2 or proto3 syntax. proto3 is
+                     the default.
+EOF
+  exit 1
+}
+
+function mkdir_if_necessary() {
+  mkdir "$@" 2>/dev/null || true
```

Comment:
Next time, you might try `mkdir -p` which succeeds if the directory already exists.